include/zephyr/drivers/sensor.h

```diff
@@ -395,13 +395,25 @@ typedef int (*sensor_sample_fetch_t)(const struct device *dev,
 typedef int (*sensor_channel_get_t)(const struct device *dev,
 				    enum sensor_channel chan,
 				    struct sensor_value *val);
+/**
+ * @typedef sensor_rgeister_get_t
+ * @brief Callback API for getting a reading register value from a sensor
+ *
+ * See sensor_rgeister_get() for argument description
+ */
+typedef int (*sensor_register_t)(const struct device *dev,
+                                    uint8_t reg,
+                                    uint8_t *val,
+				    uint16_t len);
 
 __subsystem struct sensor_driver_api {
 	sensor_attr_set_t attr_set;
 	sensor_attr_get_t attr_get;
 	sensor_trigger_set_t trigger_set;
 	sensor_sample_fetch_t sample_fetch;
 	sensor_channel_get_t channel_get;
+	sensor_register_t register_get;
```

Comment:
For what reason are we making this modification to the generic sensor driver API? For one not all sensors have registers, and (second of all) usually this functionality is achieved through sensor_attr_set/sensor_attr_get.